src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestProcessor.java

```diff
@@ -22,44 +28,110 @@
+    Case caze = getCaseByCaseId(fulfilmentRequestPayload);
 
-    Optional<Case> cazeResult = caseRepository.findByCaseId(UUID.fromString(caseId));
+    validateFulfilmentRequiredFields(
+        fulfilmentRequestEvent, fulfilmentRequestPayload, caze.getCaseId().toString());
 
-    if (cazeResult.isEmpty()) {
-      log.error(CASE_NOT_FOUND_ERROR);
-      throw new RuntimeException(String.format("Case ID '%s' not found!", caseId));
+    eventLogger.logFulfilmentRequestedEvent(
+        caze,
+        caze.getCaseId(),
+        fulfilmentRequestEvent.getDateTime(),
+        FULFILMENT_REQUEST_RECEIVED,
+        FULFILMENT_REQUESTED,
+        fulfilmentRequestPayload,
+        fulfilmentRequestEvent);
+
+    if (individualResponseRequestCodes.contains(fulfilmentRequestPayload.getFulfilmentCode())) {
+      Case individualResponseCase = saveIndividualResponseCase(caze);
+      caseProcessor.emitCaseCreatedEvent(individualResponseCase);
     }
+  }
+
+  private Case saveIndividualResponseCase(Case caze) {
+    Case individualResponseCase =
+        caze.toBuilder()
+            .caseId(UUID.randomUUID())
+            .caseRef(caseProcessor.getUniqueCaseRef())
+            .uacQidLinks(null)
+            .events(null)
+            .createdDateTime(OffsetDateTime.now())
+            .state(CaseState.ACTIONABLE)
+            .receiptReceived(false)
+            .refusalReceived(false)
+            .addressType(HOUSEHOLD_INDIVIDUAL_RESPONSE_ADDRESS_TYPE)
+            .addressLevel(null)
+            .htcWillingness(null)
+            .htcDigital(null)
+            .fieldCoordinatorId(null)
+            .fieldOfficerId(null)
+            .treatmentCode(null)
+            .ceExpectedCapacity(null)
+            .build();
+
+    caseRepository.save(individualResponseCase);
+
+    return individualResponseCase;
+  }
 
-    if (fulfilmentRequestEvent.getDateTime() == null) {
+  private void validateFulfilmentRequiredFields(
+      EventDTO eventDTO, FulfilmentRequestDTO fulfilmentRequestDTO, String caseId) {
+    if (eventDTO.getDateTime() == null) {
       log.error(DATETIME_NOT_PRESENT);
       throw new RuntimeException(
           String.format("Date time not found in fulfilment request event for Case ID '%s", caseId));
     }
 
-    Case caze = cazeResult.get();
+    if (StringUtils.isEmpty(fulfilmentRequestDTO.getFulfilmentCode())) {
+      log.error(FULFILMENT_CODE_NOT_FOUND);
+      throw new RuntimeException(
+          String.format(
+              "Fulfilment code '%s' not found from event for Case ID '%s",
+              caseId, fulfilmentRequestDTO.getFulfilmentCode()));
+    }
+  }
 
-    eventLogger.logFulfilmentRequestedEvent(
-        caze,
-        UUID.fromString(caseId),
-        fulfilmentRequestEvent.getDateTime(),
-        FULFILMENT_REQUEST_RECEIVED,
-        FULFILMENT_REQUESTED,
-        fulfilmentRequestPayload,
-        fulfilmentRequestEvent);
+  private Case getCaseByCaseId(FulfilmentRequestDTO fulfilmentRequest) {
```

Comment:
how many bazzilions of places in the code are there already similar methods as this? Also it doesn't do what it says on the method name: "find by case ID". If it's "find by case ID" then it should take "case ID" as the one and only argument

Reply:
this function is called from everywhere that does findCaseById ... now, and it's in CaseProcessor